components/node/sources/rollupclient.go

```diff
@@ -18,9 +18,9 @@ func NewRollupClient(rpc client.RPC) *RollupClient {
 	return &RollupClient{rpc}
 }
 
-func (r *RollupClient) OutputAtBlock(ctx context.Context, blockNum uint64, includeNextBlock bool) (*eth.OutputResponse, error) {
+func (r *RollupClient) OutputAtBlock(ctx context.Context, blockNum uint64, isForFaultProof bool) (*eth.OutputResponse, error) {
```

Comment:
What about seperating them into `OutputAtBlock` and `OutputAtBlockWithProof` ?

Reply:
What if `PublicInputProof` wraps all the field required for a public input and rename that flag to `includePublicInputProof`? Does it make sense?

```go
type OutputResponse struct {
	Version               Bytes32           `json:"version"`
	OutputRoot            Bytes32           `json:"outputRoot"`
	BlockRef              L2BlockRef        `json:"blockRef"`
	NextBlockRef          L2BlockRef        `json:"nextBlockRef"`
	WithdrawalStorageRoot common.Hash       `json:"withdrawalStorageRoot"`
	StateRoot             common.Hash       `json:"stateRoot"`
	Status                *SyncStatus       `json:"syncStatus"`
	PublicInputProof      *PublicInputProof `json:"publicInputProof"`
}
```

Reply:
good

Reply:
Sorry :( I don't have much time to resolve this requirement. Could somebody else do this for me..?